### DISCUSSION TITLE
Things I'm Never Going To Do

### DISCUSSION THREAD

**Discussion by ghost** [janet-lang/janet#1435] [General]
I've been in and out of software spaces for years now. I find a thing, my mind latches onto it, it spins in place until it snaps, I realise I'm doing nothing and destroying myself, I pull away, I figure one short look back can't hurt, goto 10. I hate computers, I hate software, I hate myself for obsessing over something that doesn't matter and won't affect me in any way. People go out and do things, I stay in and watch things; the only community dynamics I can see from here are ones that conduct business online, and the only remotely healthy ones of those are software. I wish I had friends. I wish I could do things.

There are some things that I've described myself as wanting to exist, but only so that they then would; I would never use them, it would just mean I could put the issues to rest and get on with something else. Of course, there will *always* be issues; making things right has been a lost cause since before most of us were born, when Jean-Louis Gassée got greedy or FreeBSD played chicken or Symbolics went bankrupt or Alonzo Church faded into obscurity.

Anyways, imaginary problems that I can't seem to banish from my head:
- AsciiDoc is *criminally* underused. Write pegn and native parser for drop-in adoption, fully compliant minus deprecated/duplicate features.
- Emulators could have fine-grained hooks. Invent entirely new API to insert eBPF into the engine to check events against conditions and signal code to run somehow.

**(abandon brain cells, all ye who venture beyond here.)**
- Janet should be the lingua franca.
    - Trash the API: put everything in a context object, make it fully incremental with no atomic phase, if such an algorithm exists.
    - Bindings to LAPACK and some graphics whatever.
    - A fully compliant JIT compiler with identical bytecode(!).
- Modern keyboard layouts (all of them) don't have hjkl under the fingers. Rewrite the vis editor to bind no keys at all except a panic button that closes it. Also make it take pegns instead of lpegs for some reason
- ELF is a bit crusty(!). Design an entirely new object file format which isn't extensible, doesn't depend on processing order, doesn't waste a single bit, and hardlinks everything to everything else. Also, a binary header format, and a minimal interface to link against in lieu of an entire shared object.

Yeeeah. Anyone heard of Redox? A while back there was a dev who wrote up a spec for the cleanest, most optimal filesystem ever conceived, then just...didn't make it. He just couldn't make peace with a rotten foundation, and the whole of IT is rotten to the core. He retreated to academia. Maybe I should do the same. Actually, no, that wouldn't help -- my brain has been spinning for weeks on a permutation problem for a game that I will never create with no ideas other than the bizarre topology of it. It's just hopeless.

Anyways, please ignore if it shouldn't be here. I just needed to get this out somewhere where people might have the slightest idea what the hell I'm on about. I hate my brain, my ADHD meds barely help because they have work enough controlling insane impulses, I wish I could either do something or beat this nonsensical compulsion to do something somewhere I'm not wanted or needed into submission.

Reply with pictures of your dogs, and shows you've watched which are philosophical but not in an "affluent alcoholic single man hates himself" kind of way.

**Comment by pepe** [comment]
It may be only me, but I take this post as a totally EJIFG. 

If the poster is still somewhere, feel free to contact me.

**Reply by tionis** [reply]
I'm not familiar with EJIFG, what does it mean?

**Reply by sogaiu** [reply]
IIRC, "Everything Janet is Fair Game"

Reference: https://github.com/janet-lang/janet/discussions/552

**Reply by tionis** [reply]
Ah I see, thanks!

**Comment by bakpakin** [comment]
While I appreciate the sentiment and empathize with the poster for the desire to create contrasted with other responsibilities and the paradox of choice, other parts of this kind of post feel a bit dark and off topic to me and a bit like a cry for help - more like a reddit venting post than a targeted discussion. Particular the talk about medication, alcoholism, self-hate, loneliness, and how "the whole of IT is rotten to the core". This is the kind of thing that, had I posted it late one night with a non-anonymous account, would probably regret in the morning. im-not-linja also has posted this sort of thing before, and I don't want it to become a pattern.

> please ignore if it shouldn't be here

Unfortunately, posting on public forums doesn't work like that.

Perhaps I have overstepped though with my mod powers - this was definitely at least partially on topic. I definitely should have given some kind of warning.

I wish the original poster the best, and encourage them to reach out to a broader community on a more appropriate forum. Wherever you are, I apologize for the abruptness - I wanted to reach out but the account seemed to be a completely anonymous account with no email and afaict there is no way to DM on github.
